kolibri/plugins/learn/assets/src/state/actions/recommended.js

```diff
@@ -0,0 +1,222 @@
+import {
+  ContentNodeResource,
+  ContentNodeProgressResource,
+  SessionResource,
+  UserExamResource,
+  ExamLogResource,
+  ExamAttemptLogResource,
+} from 'kolibri.resources';
+
+import {
+  currentUserId,
+  isFacilityUser,
+  getCurrentChannelId,
+  getCurrentChannelObject,
+} from 'kolibri.coreVue.vuex.getters';
+
+import {
+  samePageCheckGenerator,
+  setChannelInfo,
+  handleApiError,
+} from 'kolibri.coreVue.vuex.actions';
+
+import { PageNames } from '../../constants';
+import { contentState } from './main';
+
+import ConditionalPromise from 'kolibri.lib.conditionalPromise';
+import uniqBy from 'lodash/uniqBy';
+
+// User-agnostic recommendations
+
+function _getPopular(channelId) {
+  const channelPayload = { channel_id: channelId };
+  const popularPayload = { popular: 'true' };
+  return ContentNodeResource.getCollection(channelPayload, popularPayload).fetch();
```

Comment:
Could apply the _mapContentSet function in the promise chain here, but as you only do it twice below, doesn't fall under the rule of three!